`HistoryMenuItemTest.build menu items` mocks a lambda and causes intermittent failures

I saw this failure in my PR:
- GH check: https://github.com/mozilla-mobile/fenix/pull/20623/checks?check_run_id=3223771637
- Unit test artifact: https://firefoxci.taskcluster-artifacts.net/bmSV-9MZR_myzY_YYpYguQ/0/public/reports/test/testDebugUnitTest/index.html

STack trace:
```
io.mockk.MockKException: Can't instantiate proxy for class kotlin.Function1
	at io.mockk.impl.instantiation.JvmMockFactory.newProxy(JvmMockFactory.kt:64)
	at io.mockk.impl.instantiation.AbstractMockFactory.newProxy$default(AbstractMockFactory.kt:29)
	at io.mockk.impl.instantiation.AbstractMockFactory.mockk(AbstractMockFactory.kt:59)
	at org.mozilla.fenix.library.history.HistoryItemMenuTest.setup(HistoryItemMenuTest.kt:93)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
	at org.robolectric.RobolectricTestRunner$HelperTestRunner$1.evaluate(RobolectricTestRunner.java:570)
	at org.robolectric.internal.SandboxTestRunner$2.lambda$evaluate$0(SandboxTestRunner.java:278)
	at org.robolectric.internal.bytecode.Sandbox.lambda$runOnMainThread$0(Sandbox.java:89)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: io.mockk.proxy.MockKAgentException: Failed to subclass interface kotlin.jvm.functions.Function1
	at io.mockk.proxy.jvm.ProxyMaker.proxy(ProxyMaker.kt:38)
	at io.mockk.impl.instantiation.JvmMockFactory.newProxy(JvmMockFactory.kt:34)
	... 18 more
Caused by: java.lang.IllegalArgumentException: Could not create type
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:155)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:175)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation.subclass(SubclassInstrumentation.kt:51)
	at io.mockk.proxy.jvm.ProxyMaker.subclass(ProxyMaker.kt:113)
	at io.mockk.proxy.jvm.ProxyMaker.proxy(ProxyMaker.kt:35)
	... 19 more
Caused by: java.lang.IllegalStateException: Failed to invoke proxy for public abstract java.lang.Class net.bytebuddy.dynamic.loading.ClassInjector$UsingLookup$MethodHandles$Lookup.defineClass(java.lang.Object,byte[]) throws java.lang.IllegalAccessException
	at net.bytebuddy.utility.dispatcher.JavaDispatcher$ProxiedInvocationHandler.invoke(JavaDispatcher.java:1123)
	at com.sun.proxy.$Proxy61.defineClass(Unknown Source)
	at net.bytebuddy.dynamic.loading.ClassInjector$UsingLookup.injectRaw(ClassInjector.java:1621)
	at net.bytebuddy.dynamic.loading.ClassInjector$AbstractBase.inject(ClassInjector.java:113)
	at net.bytebuddy.dynamic.loading.ClassLoadingStrategy$UsingLookup.load(ClassLoadingStrategy.java:492)
	at net.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:100)
	at net.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:6147)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation.doInterceptedSubclassing(SubclassInstrumentation.kt:98)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation.access$doInterceptedSubclassing(SubclassInstrumentation.kt:21)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation$subclass$1.call(SubclassInstrumentation.kt:54)
	at io.mockk.proxy.jvm.transformation.SubclassInstrumentation$subclass$1.call(SubclassInstrumentation.kt:21)
	at net.bytebuddy.TypeCache.findOrInsert(TypeCache.java:153)
	... 23 more
Caused by: java.lang.LinkageError: loader 'app' attempted duplicate class definition for kotlin.jvm.functions.Function1$Subclass2. (kotlin.jvm.functions.Function1$Subclass2 is in unnamed module of loader 'app')
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.System$2.defineClass(System.java:2131)
	at java.base/java.lang.invoke.MethodHandles$Lookup.defineClass(MethodHandles.java:962)
	at java.base/jdk.internal.reflect.GeneratedMethodAccessor15.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at net.bytebuddy.utility.dispatcher.JavaDispatcher$Invoker$Dispatcher.invoke(Unknown Source)
	at net.bytebuddy.utility.dispatcher.JavaDispatcher$Dispatcher$ForNonStaticMethod.invoke(JavaDispatcher.java:986)
	at net.bytebuddy.utility.dispatcher.JavaDispatcher$ProxiedInvocationHandler.invoke(JavaDispatcher.java:1110)
	... 34 more

```

--- a/app/src/test/java/org/mozilla/fenix/library/history/HistoryItemMenuTest.kt
+++ b/app/src/test/java/org/mozilla/fenix/library/history/HistoryItemMenuTest.kt
@@ -6,8 +6,6 @@ package org.mozilla.fenix.library.history
 
 import android.content.Context
 import androidx.appcompat.view.ContextThemeWrapper
-import io.mockk.mockk
-import io.mockk.verify
 import mozilla.components.concept.menu.candidate.TextStyle
 import mozilla.components.support.ktx.android.content.getColorFromAttr
 import mozilla.components.support.test.robolectric.testContext
@@ -23,14 +21,16 @@ import org.mozilla.fenix.library.history.HistoryItemMenu.Item
 class HistoryItemMenuTest {
 
     private lateinit var context: Context
-    private lateinit var onItemTapped: (Item) -> Unit
     private lateinit var menu: HistoryItemMenu
+    private var onItemTappedCaptured: Item? = null
 
     @Before
     fun setup() {
         context = ContextThemeWrapper(testContext, R.style.NormalTheme)
-        onItemTapped = mockk(relaxed = true)
-        menu = HistoryItemMenu(context, onItemTapped)
+        onItemTappedCaptured = null
+        menu = HistoryItemMenu(context) {
+            onItemTappedCaptured = it
+        }
     }
 
     @Test
@@ -43,7 +43,7 @@ class HistoryItemMenuTest {
         )
 
         deleteItem.onClick()
-        verify { onItemTapped(Item.Delete) }
+        assertEquals(Item.Delete, onItemTappedCaptured)
     }
 
     @Test
@@ -59,18 +59,18 @@ class HistoryItemMenuTest {
         assertEquals("Delete", delete.text)
 
         copy.onClick()
-        verify { onItemTapped(Item.Copy) }
+        assertEquals(Item.Copy, onItemTappedCaptured)
 
         share.onClick()
-        verify { onItemTapped(Item.Share) }
+        assertEquals(Item.Share, onItemTappedCaptured)
 
         openInNewTab.onClick()
-        verify { onItemTapped(Item.OpenInNewTab) }
+        assertEquals(Item.OpenInNewTab, onItemTappedCaptured)
 
         openInPrivateTab.onClick()
-        verify { onItemTapped(Item.OpenInPrivateTab) }
+        assertEquals(Item.OpenInPrivateTab, onItemTappedCaptured)
 
         delete.onClick()
-        verify { onItemTapped(Item.Delete) }
+        assertEquals(Item.Delete, onItemTappedCaptured)
     }
 }

